### PR TITLE
feat(analytics): add event for insufficient gas warnings

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

When preparing a tx in the various flows, if it fails with insufficient gas, add an analytics event. 

### Test plan

Unit tests, manual

### Related issues

- Fixes ACT-1256

### Backwards compatibility

Yes

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [x] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/earn/EarnCollectScreen.tsx [hunk 1]
```diff
@@ -58,6 +58,19 @@ export default function EarnCollectScreen({ route }: Props) {
     depositTokenId,
     feeCurrencies,
   })
+
+  useEffect(() => {
+    if (
+      asyncPreparedTransactions.result?.type === 'need-decrease-spend-amount-for-gas' ||
+      asyncPreparedTransactions.result?.type === 'not-enough-balance-for-gas'
+    ) {
```
Comment (MuckT) [inline src/earn/EarnCollectScreen.tsx:66]:
```suggestion
    if (
      ['need-decrease-spend-amount-for-gas', 'not-enough-balance-for-gas']
      .includes(asyncPreparedTransactions.result?.type)
    ) {
```

File: src/send/EnterAmount.test.tsx [hunk 2]
```diff
@@ -126,6 +126,7 @@ const defaultParams = {
   prepareTransactionError: undefined,
   onPressProceed: onPressProceedSpy,
   ProceedComponent: SendProceed,
+  origin: 'send' as const,
```
Comment (MuckT) [inline src/send/EnterAmount.test.tsx:129]: Is this type cast needed?

File: src/walletConnect/saga.test.ts [hunk 3]
```diff
@@ -570,6 +572,64 @@ describe('showActionRequest', () => {
       preparedTransaction: mockPreparedTransactions.transactions[0],
       prepareTransactionErrorMessage: undefined,
     })
+    expect(ValoraAnalytics.track).toHaveBeenCalledTimes(1)
+    expect(ValoraAnalytics.track).toHaveBeenCalledWith(
+      WalletConnectEvents.wc_request_propose,
+      expect.anything()
```
Comment (MuckT) [inline src/walletConnect/saga.test.ts:578]: Is this because we could have an array of any number of transactions?

**Review by MuckT — APPROVED** [review]

File: src/send/EnterAmount.test.tsx (hunk 2 above)
Comment (satish-ravi) [reply src/send/EnterAmount.test.tsx:129]: yes, otherwise TS infers it as a generic string.

File: src/walletConnect/saga.test.ts (hunk 3 above)
Comment (satish-ravi) [reply src/walletConnect/saga.test.ts:578]: this was just me being lazy, it wasn't super obvious what the expected props were, there were too many 😄

**Review by kathaypacific — COMMENTED** [review]
without thinking deeply about how every screen works, it's hard to say if that useEffect will be rerun unexpectedly especially because there are multiple dependencies in the dependency array. e.g. you might have a prepared transaction that has insufficient gas, then change the token / network id (idk if possible) - if the network id is in the dep array then you're going to get another event triggered, even though the new tx hasn't been prepared yet...again, not sure if this is a real scenario / whether we care about duplicate events. 

did you consider dispatching the event in prepareTransactions directly? it seems easy to forget to add this useEffect block when implementing new screens, so it'd be nice to do this somewhere central

**Comment by satish-ravi** [comment]
@kathaypacific good call, I think if network id changes, we clear the prepared tx but yeah there might be races. I originally added it per flow to have more control per flow on when to fire the event, but the behavior is same as firing from the prepareTransactions method. Made #5671 instead

**Pull request closed without merging**